yarl/_url.py

```diff
@@ -32,8 +32,8 @@
 
 import idna
 from multidict import MultiDict, MultiDictProxy, istr
+from propcache import under_cached_property as cached_property
```

Comment:
Hoping the with the https://github.com/aio-libs/propcache/issues/18 proposal, this could become
```suggestion
from propcache.api import under_cached_property as cached_property
```